.gitmodules

```diff
@@ -7,3 +7,12 @@
 [submodule "libTO"]
 	path = iotc-azrtos-sdk/deps/libTO
 	url = https://github.com/TrustedObjects/libTO.git
+[submodule "threadx"]
+	path = os/threadx
+	url = https://github.com/azure-rtos/threadx.git
+[submodule "netxduo"]
+	path = os/netxduo
+	url = https://github.com/azure-rtos/netxduo.git
```

Comment:
> I guess I am missing a lot - not understanding how can this be mapped to a specific Azure RTOS release (containing demo files, glue and platform-specific files and libraries). Last I checked, that netxduo git was missing ports and other OS specific. Maybe we could chat so you can explain how this works.

Hi Nik, good question. So my solution to this was to commit the changed files directly into the sample. Then when we generate the symlinks, we only link the files that aren't currently in the sample. Looking at the PR you can see nxd_dns.c and nx_user.h  have been added to the sample.

Reply:
I'm not interested in how does the github submodule and linking work. 

What I mean is - other projects released by Microsoft  have a lot more value than what's on the main branch of netx, for example. for example, ADU drivers, WiFi drivers (and how it hooks into NetX. Maybe Renesas only has mainly pure netx form the main branch and only a handful of glue files? Just do a clone of whatever same54xprov2 depends on and compare it's netx directory to main branc on their netx github repo. It's an 90% intersect, but there are files which exist only on one or the other side.

The problem that I probably have is with the idea of not using Microsoft AzureRTOS release as a baseline. I would think that we should be using their own release as a baseline where ever possible because Microsoft or chip vendors would be doing testing on a set of files. We should depend on those files (bar our own overrides) and should not be putting together a set of component releases (filex/netx/threadx).

In addition, their set of released files contains a sample that we can use to check if something that doesn't work with our SDK potentially work.

I'd want to hear your opinion on this. I'm sure it's not just about making the project use gitmodules instead of zipped release because someone said it should be so.

Reply:
I can see where your coming from here. To be honest I think the RX65N fits this structure much better than other projects. After porting over the STM32L4 build to use these submodules (and more besides), its a lot less clean than just using the RTOS port/BSP from the zipped project files. The Git repos often have much more than is needed, and I'm not sure there's a way to selectively clone them per project. I could imagine it might make for quite a bloated environment after a few projects. If other projects are worse, then the juice might not be worth the squeeze.

How about a compromise? We use submodules only for the iotc-azure-rtos dependencies (iotc-c-lib, cJSON, libTO (?)). That way the user only has to clone them once. I'll create a new branch so you can see what you think.

Reply:
Regarding iotc-c-lib, cJSON, libTO, I absolutely agree that git submodules approach is better. 

Also, we have the two new Microchip projects where Azure RTOS baseline is not in a zip, but in git repo. I was planning on converting those to git modules. The downside to doing this would be that we would be unnecessarily adding to the time to clone a bunch of dependencies that are potentially not needed for the project that the user is interested in. I'd be interested to hear your opinion on that.

What you saw when trying to integrate the git repos for STM32 and similar was what I was worried about. It's extra integration work for very little benefit. So let's please stick with the zip pull then. 

The thing that is still useful is that you learned how to integrate git netx into the projects. We had cases where we needed to upgrade baseline netx (for decure element support). I could not upgrade easily using the netx git repo baseline. Fortunately, the new full Azure RTOS release contained what I needed and I used that.